datadog_checks_base/tests/base/checks/openmetrics/test_v2/test_transformers/test_counter.py

```diff
@@ -11,52 +11,38 @@
 
 
 def test_basic(aggregator, dd_run_check, mock_http_response):
+    """
+    Metrics with type 'counter' in the raw payload get collected.
+
+    If they end in '_total' that suffix is dropped.
+    """
+
     mock_http_response(
         """
-        # HELP go_memstats_alloc_bytes_total Total number of bytes allocated, even if freed.
-        # TYPE go_memstats_alloc_bytes_total counter
-        go_memstats_alloc_bytes_total 9.339544592e+09
-        # HELP go_memstats_frees_total Total number of frees.
-        # TYPE go_memstats_frees_total counter
-        go_memstats_frees_total 1.28219257e+08
+        # HELP foo_total Example of '_total' getting dropped
+        # TYPE foo_total counter
+        foo_total 9.339544592e+09
+        # HELP bar_count Example of '_count' not getting dropped
+        # TYPE bar_count counter
+        bar_count 1.28219257e+08
+        # HELP baz Example that doesn't end in '_total' nor '_count'
+        # TYPE baz counter
+        baz 1.28219257e+08
         """
     )
     check = get_check({'metrics': ['.+']})
     dd_run_check(check)
 
     aggregator.assert_metric(
-        'test.go_memstats_alloc_bytes.count', 9339544592, metric_type=aggregator.MONOTONIC_COUNT, tags=['endpoint:test']
+        'test.foo.count', 9339544592, metric_type=aggregator.MONOTONIC_COUNT, tags=['endpoint:test']
     )
     aggregator.assert_metric(
-        'test.go_memstats_frees.count', 128219257, metric_type=aggregator.MONOTONIC_COUNT, tags=['endpoint:test']
+        'test.bar_count.count', 128219257, metric_type=aggregator.MONOTONIC_COUNT, tags=['endpoint:test']
```

Comment:
I don't understand why '_total' should be dropped but not '_count'.

Reply:
It was to keep parity with the prometheus_client:
https://github.com/prometheus/client_python/blob/eb4ac52da44b587293965723079b6104e200e5b0/README.md#counter